packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -413,6 +418,47 @@ class CupertinoDatePicker extends StatefulWidget {
   /// Defaults to a value that matches the default iOS date picker wheel.
   final double itemExtent;
 
+  /// A function that returns a widget that overlaid on the picker
+  /// to highlight the currently selected entry.
+  ///
+  /// If unspecified, it defaults to a [CupertinoPickerDefaultSelectionOverlay]
+  /// which is a gray rounded rectangle overlay in iOS 14 style.
+  ///
+  /// If the selection overlay builder returns null, no overlay will be drawn.
+  ///
+  /// ## Example
+  ///
+  /// This example shows how to recreate the default selection overlay
+  /// with selectionOverlayBuilder.
+  ///
+  /// ```dart
```

Comment:
I would probably make this a snippet. You can see an example of this in:

https://github.com/flutter/flutter/blob/586e1fce1b25f7115b5e6c813047e7c0fbd8a796/packages/flutter/lib/src/material/tab_controller.dart#L30

And you can see what that does in the docs site: https://main-api.flutter.dev/flutter/material/TabController-class.html